src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -95,6 +96,8 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 		this._register(this._capabilities.onDidAddCapability(c => {
 			if (c === TerminalCapability.CommandDetection) {
 				this._addCommandDetectionListeners();
+			} else if (c === TerminalCapability.BufferMarkDetection) {
+				this._addBufferMarkListeners();
 			}
 		}));
 		this._register(this._capabilities.onDidRemoveCapability(c => {
```

Comment:
Even though it cannot happen currently, we should handle removal whenever adding is handled. This could be done fairly generically with a map:

```ts
this._capabilityDisposables.add(c, ...)
...
dispose(this._capabilityDisposables.get?(c));
this._capabilityDisposables.delete(c);
```